File: matiasmasso/Mat.NET/DTO/DTOGuidNom.cs
using System;

namespace DTO
{
    public class DTOGuidNom : DTOBaseGuid
    {
        public string Nom { get; set; }
        // Property Nom As String

        public DTOGuidNom(Guid oGuid, string sNom = "") : base(oGuid)
        {
            Nom = sNom;
        }

        public DTOGuidNom() : base()
        {
        }

        public static DTOGuidNom Factory(Guid oGuid, string sNom = "")
        {
            var retval = new DTOGuidNom(oGuid);
            {
                retval.Nom = sNom;
            }
            return retval;
        }

        public Compact CompactCustomer()
        {
            Compact retval = Compact.Factory(this.Guid, this.Nom);
            return retval;
        }

        public class Compact
        {
            public Guid Guid { get; set; }
            public String Nom { get; set; }

            public static Compact Factory(Guid guid, String nom = "")
            {
                Compact retval = new Compact();
                retval.Guid = guid;
                retval.Nom = nom;
                return retval;
            }

            public bool Equals(DTOBaseGuid oCandidate)
            {
                bool retval = false;
                if (oCandidate != null)
                    retval = Guid.Equals(oCandidate.Guid);
                return retval;
            }
        }
    }
}

File: matiasmasso/Mat.NET/DT2/DTOGuidNom.vb
Public Class DTOGuidNom

    Inherits DTOBaseGuid
    Property nom As String
    'Property Nom As String

    Public Sub New(oGuid As Guid, Optional sNom As String = "")
        MyBase.New(oGuid)
        _Nom = sNom
    End Sub

    Public Sub New()
        MyBase.New()
    End Sub

    Shared Function Factory(oGuid As Guid, Optional sNom As String = "") As DTOGuidNom
        Dim retval = New DTOGuidNom(oGuid)
        With retval
            .Nom = sNom
        End With
        Return retval
    End Function
End Class
